ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/API.hs

```diff
@@ -207,6 +207,24 @@ data ChainDB m blk = ChainDB {
       -- non-deterministic.
     , getIsFetched       :: STM m (Point blk -> Bool)
 
+      -- | Return a function that checks whether a block has been validated
+      -- and found to be valid.
+      --
+      -- The function will return 'False' for blocks that were found to be
+      -- invalid, blocks that haven't been validated yet (because they are
+      -- part of a short fork or are disconnected from the current chain), or
+      -- blocks older than @k@, i.e., blocks no longer in the volatile DB. For
+      -- blocks older than that the results should be regarded as
+      -- non-deterministic.
```

Comment:
Turn this into a list and replace the last one with: "Blocks older than `k` may or may not be in the domain".